<a href="https://colab.research.google.com/github/Nikolai-N484/Data201_NikolaiN/blob/main/Project2/Nikolai_Navarro_Project2_Week11.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Data 201 Project 2**
## **Nikolai Navarro**
## Spring 2026
---

## **Introduction**

Provide a brief description of:

- Dataset source  
- Population of interest  
- Data collection method (if known)  
- Limitations of the data  
- Possible bias or ethical concerns  

Include a citation for your dataset.

The dataset I am using was collected by the U.S. Census Bureau in 2000. It contains information about 500 random United States citizens such as their age, income, race, and sex. One limitation of the data is that it contains information from over 2 decades ago but it will still be interesting to look at.

Citation: “Random Sample of 2000 U.S. Census Data.” OpenIntro, U.S. Census Bureau, www.openintro.org/data/index.php?data=census.

---
## **Data Overview**

Describe your dataset:

- Number of rows and columns  
- What each row represents  
- Table of variables:
  - Name  
  - Type (quantitative/categorical)  
  - Description  

Include any data cleaning or transformations performed.

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv("https://raw.githubusercontent.com/Nikolai-N484/Data201_NikolaiN/refs/heads/main/Project2/census.csv")

df.head()

,census_year,state_fips_code,total_family_income,age,sex,race_general,marital_status,total_personal_income
0,2000,Florida,14550.0,44,Male,Two major races,Married/spouse present,0.0
1,2000,Florida,22800.0,20,Female,White,Never married/single,13000.0
2,2000,Florida,0.0,20,Male,Black,Never married/single,20000.0
3,2000,Florida,23000.0,6,Female,White,Never married/single,NaN
4,2000,Florida,48000.0,55,Male,White,Married/spouse present,36000.0


In [3]:
columns = df.columns

shape = df.shape

print(columns)
print(shape)

Index(['census_year', 'state_fips_code', 'total_family_income', 'age', 'sex',
       'race_general', 'marital_status', 'total_personal_income'],
      dtype='object')
(500, 8)


In [4]:
count_nan = df.isna().sum()
print(count_nan)

census_year                0
state_fips_code            0
total_family_income       15
age                        0
sex                        0
race_general               0
marital_status             0
total_personal_income    108
dtype: int64


In [14]:
df2 = df.dropna()
count_nan2 = df2.isna().sum()
shape2 = df2.shape

print(count_nan2)
print(shape2)

census_year              0
state_fips_code          0
total_family_income      0
age                      0
sex                      0
race_general             0
marital_status           0
total_personal_income    0
dtype: int64
(377, 8)


In [15]:
df2.head()

,census_year,state_fips_code,total_family_income,age,sex,race_general,marital_status,total_personal_income
0,2000,Florida,14550.0,44,Male,Two major races,Married/spouse present,0.0
1,2000,Florida,22800.0,20,Female,White,Never married/single,13000.0
2,2000,Florida,0.0,20,Male,Black,Never married/single,20000.0
4,2000,Florida,48000.0,55,Male,White,Married/spouse present,36000.0
5,2000,Florida,74000.0,43,Female,White,Married/spouse present,27000.0


After removing the data with NA's this data set contains a total of 377 rows and 8 columns.

| Column | Description | Data Type |
|:-----|:------:|------:|
|  census_year  |  Census Year  |  Quantitative  |
|  state_fips_code  |  Name of state  |  Categorical  |
|  total_family_income  |  Total family income (in U.S. dollars)  |  Quantitative  |
|  age  |  Persons Age  |  Quantitative  |
|  sex  |  Persons Sex  |  Categorical  |
|  race_general  |  Persons Race  |  Categorical  |
|  marital_status  |  Persons Marital Status  |  Categorical  |
|  total_personal_income  |  Total personal income (in U.S. dollars)  |  Quantitative  |

---
## 4. **Data Analysis and Modeling**

### a. Data Exploration and Visualization

Create and interpret **at least 3 visualizations**, such as:

- Histograms  
- Scatter plots  
- Boxplots  
- Bar charts  

Describe **patterns, relationships, or interesting observations**.


### b. Train/Test Split  

Split your data into:

- Training set  
- Testing set  

Explain your choice (e.g., 80/20 split).

### c. Build a Baseline Model  

Build a **linear regression model** using your selected variables.

Explain:

- What variables you included  
- Why you chose them

### d. Improve Your Model  

Modify your model using at least **ONE** of the following:

- Transformation (log, polynomial, etc.)  
- Interaction term  
- Feature selection (adding/removing variables)  

You must:

- Explain **why** you made the change  
- Support your decision using:
  - plots  
  - patterns in the data  
  - model results

### e. Compare Models  

Compare your baseline and improved models using appropriate metrics:

- RMSE  
- \(R^2\)

Clearly explain:

- Which model performs better  
- Why

### f. Interpret Your Final Model  

Explain your regression model in plain language:

- Which variables are important?  
- What do the coefficients mean?  
- How do predictors affect the response variable?

### g. Predictions  

Use your final model to:

- Make predictions for **3 new data points**  
- Interpret the results in context

---
## **Conclusions and Recommendations**

Summarize:

- Key findings  
- Model performance  
- Any challenges or limitations  
- Whether your model generalizes well  
- Possible next steps or improvements  

If results are not strong or interesting, explain why.